components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-list.jsp

```diff
@@ -30,166 +50,267 @@
+                                 for (IdentityProvider idp : idpsToDisplay) {
+                                         boolean enable = idp.getEnable();
+                                 %>
+                              <tr>
+                                 <td><%=Encode.forHtmlContent(idp.getIdentityProviderName())%></td>
+                                 <td><%=idp.getIdentityProviderDescription() != null ? Encode.forHtmlContent(idp.getIdentityProviderDescription()) : ""%></td>
+                                 <td style="width: 100px; white-space: nowrap;">
+                                    <% if (enable) { %>
+                                    <a title="<fmt:message key='disable.policy'/>"
+                                       onclick="
+                                       enableOrDisableIdP('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>', 0);return false;"
+                                       href="#" style="background-image: url(images/disable.gif);" class="icon-link">
+                                       <fmt:message key='disable.policy'/>
+                                    </a>
+                                    <% } else { %>
+                                    <a title="<fmt:message key='enable.policy'/>"
+                                       onclick="enableOrDisableIdP('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>', 1);return false;"
+                                       href="#" style="background-image: url(images/enable2.gif);" class="icon-link">
+                                       <fmt:message key='enable.policy'/>
+                                    </a>
+                                    <% } %>
                                     <a title="<fmt:message key='edit.idp.info'/>"
-                                       onclick="editIdPName('<%=Encode.forJavaScriptAttribute(identityProvidersList.get(i).getIdentityProviderName())%>');return false;"
-                                       href="#"
-                                       class="icon-link"
-                                       style="background-image: url(images/edit.gif)">
+                                       onclick="editIdPName('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>');return false;"
+                                       style="background-image: url(images/edit.gif);" class="icon-link">
                                        <fmt:message key='edit'/>
                                     </a>
                                     <a title="<fmt:message key='delete'/>"
-                                       onclick="deleteIdPName('<%=Encode.forJavaScriptAttribute(identityProvidersList.get(i).getIdentityProviderName())%>');return false;"
+                                       onclick="deleteIdPName('<%=Encode.forJavaScriptAttribute(idp.getIdentityProviderName())%>', '<%=pageNumberInt%>', '<%=Encode.forJavaScriptAttribute(filterString)%>');return false;"
```

Comment:
since pageNumber is a request parameter let's encode that

Reply:
We do not want to encode while passing the integer parameter